twotp/term.py

```diff
@@ -205,6 +205,35 @@ def Set(val):
     return set(val)
 
 
+class Map(Term):
+    def __init__(self, val):
+        self._val = val
+
+
+    def __len__(self):
```

Comment:
It'd be worth maybe inheriting collections.Mapping and to provide the necessary methods.